java/maven.indexer/src/org/netbeans/modules/maven/indexer/NexusRepositoryIndexerImpl.java

```diff
@@ -417,7 +420,11 @@ private boolean loadIndexingContext(final RepositoryInfo info) throws IOExceptio
         }
     }
 
-    private @CheckForNull IteratorSearchResponse repeatedPagedSearch(Query q, final List<IndexingContext> contexts, int count) throws IOException {
+    private @CheckForNull IteratorSearchResponse repeatedPagedSearch(Query q, IndexingContext context, int count) throws IOException {
+        return repeatedPagedSearch(q, List.of(context), count);
```

Comment:
Is using `List#of` safe here? If I remember correctly `Collections#singletonList` can handle `null`, while `List#of` will throw a `NullPointerException`.

Reply:
context is not allowed to be null. This serves also as early null check.

If it would be null it would cause an exception in maven-indexer's [`DefaultSearchEngine`](https://github.com/apache/maven-indexer/blob/fbf7c0c7d305756092ba93e40d77c0a3f0ce7eab/indexer-core/src/main/java/org/apache/maven/index/DefaultSearchEngine.java#L328).